### PR TITLE
Add OS version to download fail pixel

### PR DESCRIPTION
**Pull request by jotaemepereira**
Task/Issue URL: https://app.asana.com/0/1204006570077678/1208780258680355/f
Tech Design URL:
CC:

**Description**:
It adds the OS version to this pixel to show where we are experiencing the download issue.

**Steps to test this PR**:
1. You can add a throw in the 323 line
2. Check the `fileDownloadCreatePresentersFailed` pixel has the `OSVersion` (it should be in the logs)

**Definition of Done**:

* [x] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

—
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: DuckDuckGo/Statistics/GeneralPixel.swift [hunk 1]
```diff
@@ -1297,6 +1297,8 @@ enum GeneralPixel: PixelKitEventV2 {
                 parameters[NewTabSearchBoxExperimentPixel.Parameters.onboardingCohort] = onboardingCohort.rawValue
             }
             return parameters
+        case .fileDownloadCreatePresentersFailed(let OSVersion):
+            return ["OSVersion": OSVersion]
```
Comment (ayoy) [inline DuckDuckGo/Statistics/GeneralPixel.swift:1301]: Can we call this "osVersion"? Some of other parameters are like this:
```swift
        public static let appVersion = "appVersion"
        public static let pixelSource = "pixelSource"
        public static let osMajorVersion = "osMajorVersion"
```

Also the associated parameter of the enum value should be `osVersion`, per convention.

Comment (jotaemepereira) [reply DuckDuckGo/Statistics/GeneralPixel.swift:1301]: Yes, sure!

Comment (jotaemepereira) [reply DuckDuckGo/Statistics/GeneralPixel.swift]: Done ✅

**Review by ayoy — APPROVED** [review]
LGTM 👍

**Pull request merged**